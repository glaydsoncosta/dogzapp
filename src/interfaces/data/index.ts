// Data types
interface Breed {
  name: string;
  types: string[];
}

interface Breeds {
  list: Breed[];
}

interface BreedImages {
  message: string[];
}

// API responses
interface BreedsApiResponse {
  message: {};
}

interface BreedImagesApiResponse {
  message: string[];
}

export type { Breed, Breeds, BreedImages, BreedsApiResponse, BreedImagesApiResponse };
