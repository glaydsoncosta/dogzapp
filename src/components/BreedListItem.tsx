import React from 'react';
import { StyleSheet, Text, TouchableWithoutFeedback, View } from 'react-native';
import { gray_500, gray_800, primary } from '../styles/colors';
import { Breed } from '../interfaces/data';
import FastImage from 'react-native-fast-image';
import { assets } from '../assets';

type BreedListItemProps = {
  breed: Breed;
  onPress: (breed: Breed) => void;
};

export default function BreedListItem(props: BreedListItemProps) {
  const { breed, onPress } = props;

  return (
    <TouchableWithoutFeedback onPress={() => onPress(breed)}>
      <View style={styles.mainContainer}>
        <View style={styles.leftContainer}>
          <FastImage source={assets.icons.misc.breeds} resizeMode={FastImage.resizeMode.contain} style={styles.listItemIcon} tintColor={primary} />
        </View>
        <View style={styles.centerContainer}>
          <Text style={styles.breedTitleText}>{breed.name}</Text>
        </View>
        <View style={styles.rightContainer}>
          <FastImage source={assets.icons.misc.chevron} resizeMode={FastImage.resizeMode.contain} style={styles.chevronIcon} tintColor={gray_800} />
        </View>
      </View>
    </TouchableWithoutFeedback>
  );
}

const styles = StyleSheet.create({
  mainContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingVertical: 20
  },
  breedTitleText: {
    fontSize: 18,
    fontWeight: '400',
    marginLeft: 10
  },
  listItemIcon: {
    width: 32,
    height: 32
  },
  chevronIcon: {
    width: 18,
    height: 18
  },
  leftContainer: {
    width: '10%'
  },
  centerContainer: {
    width: '80%'
  },
  rightContainer: {
    width: '10%',
    justifyContent: 'center',
    alignItems: 'flex-end'
  }
});
