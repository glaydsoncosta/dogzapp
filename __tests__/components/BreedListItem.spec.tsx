import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import BreedListItem from '../../src/components/BreedListItem';
import { Breed } from '../../src/interfaces/data';

const testBreed = {
  name: 'Test',
  types: []
} as Breed;

test('Should render BreedListItem component', () => {
  const snapShot = renderer.create(<BreedListItem breed={testBreed} onPress={() => {}} />).toJSON();
  expect(snapShot).toMatchSnapshot();
});
