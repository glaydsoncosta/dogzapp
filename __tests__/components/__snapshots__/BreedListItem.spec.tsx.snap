// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render BreedListItem component 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingVertical": 20,
    }
  }
>
  <View
    style={
      {
        "width": "10%",
      }
    }
  >
    <View
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "height": 32,
            "width": 32,
          },
        ]
      }
    >
      <FastImageView
        defaultSource={null}
        resizeMode="contain"
        source={
          {
            "testUri": "../../../src/assets/icons/pawprint.png",
          }
        }
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        tintColor="#673AB7"
      />
    </View>
  </View>
  <View
    style={
      {
        "width": "80%",
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 18,
          "fontWeight": "400",
          "marginLeft": 10,
        }
      }
    >
      Test
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "flex-end",
        "justifyContent": "center",
        "width": "10%",
      }
    }
  >
    <View
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "height": 18,
            "width": 18,
          },
        ]
      }
    >
      <FastImageView
        defaultSource={null}
        resizeMode="contain"
        source={
          {
            "testUri": "../../../src/assets/icons/chevron.png",
          }
        }
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        tintColor="#424242"
      />
    </View>
  </View>
</View>
`;
